modules/host/negotiateformcontract.go

```diff
@@ -212,7 +212,8 @@ func (h *Host) managedVerifyNewContract(txnSet []types.Transaction, renterPK cry
 	blockHeight := h.blockHeight
 	lockedStorageCollateral := h.financialMetrics.LockedStorageCollateral
 	publicKey := h.publicKey
-	settings := h.settings
+	eSettings := h.externalSettings()
```

Comment:
I think here you want to be using the externalSettings that you provided at the beginning of the RPC, instead of asking for the external settings from the host again, as the value may have changed.